### ISSUE TITLE
Is this a normal case?

### ISSUE THREAD

**Issue by mrk1992** [yuhuixu1993/PC-DARTS#31]
I run this command

`python train_search.py`

and get a these results

![image](https://user-images.githubusercontent.com/11848064/74133010-fd495380-4c2a-11ea-8961-3c47c14ca148.png)

is it right ?

**Comment by yuhuixu1993** [comment]
@mrk1992 ,hi, what is your running environmrnt? You may refer to this issue: https://github.com/yuhuixu1993/PC-DARTS/issues/5

**Comment by mrk1992** [comment]
Summary
1. erase main def 
2. put parser group code and main def code into a     `if __name__ == '__main__':`
------------------------------------------------------------------------------------------

Thank you for reply. I just wanted to check if it was the problem.

I'm running the project on Windows 10.
The rest is almost the same. 

In the train_search.py file

I edit the code like,
    
    if __name__ == '__main__':
        parser = argparse.ArgumentParser("cifar")
        parser.add_argument('--data', type=str, default='./data', help='location of the data corpus')
        parser.add_argument('--set', type=str, default='cifar10', help='location of the data corpus')
        parser.add_argument('--batch_size', type=int, default=64, help='batch size') # default batch_size 256
        parser.add_argument('--learning_rate', type=float, default=0.1, help='init learning rate')
        parser.add_argument('--learning_rate_min', type=float, default=0.0, help='min learning rate')
        parser.add_argument('--momentum', type=float, default=0.9, help='momentum')
        parser.add_argument('--weight_decay', type=float, default=3e-4, help='weight decay')
        parser.add_argument('--report_freq', type=float, default=50, help='report frequency')
        parser.add_argument('--gpu', type=int, default=0, help='gpu device id')
        parser.add_argument('--epochs', type=int, default=50, help='num of training epochs') # default epochs 50
        parser.add_argument('--init_channels', type=int, default=16, help='num of init channels')
        parser.add_argument('--layers', type=int, default=8, help='total number of layers')
        parser.add_argument('--model_path', type=str, default='saved_models', help='path to save the model')
        parser.add_argument('--cutout', action='store_true', default=False, help='use cutout')
        parser.add_argument('--cutout_length', type=int, default=16, help='cutout length')
        parser.add_argument('--drop_path_prob', type=float, default=0.3, help='drop path probability')
        parser.add_argument('--save', type=str, default='EXP', help='experiment name')
        parser.add_argument('--seed', type=int, default=2, help='random seed')
        parser.add_argument('--grad_clip', type=float, default=5, help='gradient clipping')
        parser.add_argument('--train_portion', type=float, default=0.5, help='portion of training data')
        parser.add_argument('--unrolled', action='store_true', default=False, help='use one-step unrolled validation loss')
        parser.add_argument('--arch_learning_rate', type=float, default=6e-4, help='learning rate for arch encoding')
        parser.add_argument('--arch_weight_decay', type=float, default=1e-3, help='weight decay for arch encoding')
        args = parser.parse_args()

        args.save = 'search-{}-{}'.format(args.save, time.strftime("%Y%m%d-%H%M%S"))
        utils.create_exp_dir(args.save, scripts_to_save=glob.glob('*.py'))

        log_format = '%(asctime)s %(message)s'
        logging.basicConfig(stream=sys.stdout, level=logging.INFO, format=log_format, datefmt='%m/%d %I:%M:%S %p')
        fh = logging.FileHandler(os.path.join(args.save, 'log.txt'))
        fh.setFormatter(logging.Formatter(log_format))
        logging.getLogger().addHandler(fh)


        CIFAR_CLASSES = 10
        if args.set=='cifar100':
                CIFAR_CLASSES = 100

        if not torch.cuda.is_available():
                logging.info('no gpu device available')
                sys.exit(1)

        np.random.seed(args.seed)
        torch.cuda.set_device(args.gpu)
        cudnn.benchmark = True
        torch.manual_seed(args.seed)
        cudnn.enabled=True
        torch.cuda.manual_seed(args.seed)
        logging.info('gpu device = %d' % args.gpu)
        logging.info("args = %s", args)

        criterion = nn.CrossEntropyLoss()
        criterion = criterion.cuda()
        model = Network(args.init_channels, CIFAR_CLASSES, args.layers, criterion)
        model = model.cuda()
        logging.info("param size = %fMB", utils.count_parameters_in_MB(model))

        optimizer = torch.optim.SGD(model.parameters(), args.learning_rate, momentum=args.momentum, weight_decay=args.weight_decay)

        train_transform, valid_transform = utils._data_transforms_cifar10(args)
        if args.set=='cifar100':
                train_data = dset.CIFAR100(root=args.data, train=True, download=True, transform=train_transform)
        else:
                train_data = dset.CIFAR10(root=args.data, train=True, download=True, transform=train_transform)

        num_train = len(train_data)
        indices = list(range(num_train))
        split = int(np.floor(args.train_portion * num_train))

        train_queue = torch.utils.data.DataLoader(train_data, batch_size=args.batch_size, sampler=torch.utils.data.sampler.SubsetRandomSampler(indices[:split]), pin_memory=True, num_workers=2)

        valid_queue = torch.utils.data.DataLoader(train_data, batch_size=args.batch_size, sampler=torch.utils.data.sampler.SubsetRandomSampler(indices[split:num_train]), pin_memory=True, num_workers=2)

        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, float(args.epochs), eta_min=args.learning_rate_min)

        architect = Architect(model, args)

        for epoch in range(args.epochs):
                scheduler.step()
                lr = scheduler.get_lr()[0]
                logging.info('epoch %d lr %e', epoch, lr)

                genotype = model.genotype()
                logging.info('genotype = %s', genotype)

                #print(F.softmax(model.alphas_normal, dim=-1))
                #print(F.softmax(model.alphas_reduce, dim=-1))

                # training
                train_acc, train_obj = train(train_queue, valid_queue, model, architect, criterion, optimizer, lr,epoch)
                logging.info('train_acc %f', train_acc)

                # validation
                if args.epochs-epoch<=1:
                valid_acc, valid_obj = infer(valid_queue, model, criterion)
                logging.info('valid_acc %f', valid_acc)

                utils.save(model, os.path.join(args.save, 'weights.pt'))
